test/e2e_node/memory_eviction_test.go

```diff
@@ -33,70 +31,49 @@ import (
-			// Inside the func, we check that all pods are in a valid phase with
-			// respect to the eviction order of best effort, then burstable, then guaranteed.
-			By("polling the Status.Phase of each pod and checking for violations of the eviction order.")
+		AfterEach(func() {
+			// Check available memory after condition disappears, just in case:
+			// Wait for available memory to decrease to a reasonable level before ending the test.
+			// This helps prevent interference with tests that start immediately after this one.
+			By("waiting for available memory to decrease to a reasonable level before ending the test.")
 			Eventually(func() error {
-
-				gteed, gtErr := f.Client.Pods(f.Namespace.Name).Get(guaranteed.Name)
-				framework.ExpectNoError(gtErr, fmt.Sprintf("getting pod %s", guaranteed.Name))
-				gteedPh := gteed.Status.Phase
-
-				burst, buErr := f.Client.Pods(f.Namespace.Name).Get(burstable.Name)
-				framework.ExpectNoError(buErr, fmt.Sprintf("getting pod %s", burstable.Name))
-				burstPh := burst.Status.Phase
-
-				best, beErr := f.Client.Pods(f.Namespace.Name).Get(besteffort.Name)
-				framework.ExpectNoError(beErr, fmt.Sprintf("getting pod %s", besteffort.Name))
-				bestPh := best.Status.Phase
-
-				glog.Infof("pod phase: guaranteed: %v, burstable: %v, besteffort: %v", gteedPh, burstPh, bestPh)
-
-				if bestPh == api.PodRunning {
-					Expect(burstPh).NotTo(Equal(api.PodFailed), "burstable pod failed before best effort pod")
-					Expect(gteedPh).NotTo(Equal(api.PodFailed), "guaranteed pod failed before best effort pod")
-				} else if burstPh == api.PodRunning {
-					Expect(gteedPh).NotTo(Equal(api.PodFailed), "guaranteed pod failed before burstable pod")
+				memoryLimit := getMemoryLimit(f)
+				summary, err := getNodeSummary()
+				if err != nil {
+					return err
+				}
+				if summary.Node.Memory.AvailableBytes == nil {
+					return fmt.Errorf("summary.Node.Memory.AvailableBytes was nil, cannot get memory stats.")
 				}
+				avail := int64(*summary.Node.Memory.AvailableBytes)
+
+				halflimit := memoryLimit.Value() / 2
 
-				// When both besteffort and burstable have been evicted, the test has completed.
-				if bestPh == api.PodFailed && burstPh == api.PodFailed {
+				// Wait for at least half of memory limit to be available
+				if avail >= halflimit {
 					return nil
 				}
-				return fmt.Errorf("besteffort and burstable have not yet both been evicted.")
-
-			}, 60*time.Minute, 5*time.Second).Should(BeNil())
+				return fmt.Errorf("current available memory is: %d bytes. Expected at least %d bytes available.", avail, halflimit)
+			}, 5*time.Minute, 15*time.Second).Should(BeNil())
+		})
+		var guaranteed *api.Pod
+		It("should contain node memory pressure condition)", func() {
+			memoryLimit := getMemoryLimit(f)
+			By("creating a memory hogging pod.")
+			guaranteed = createMemhogPod(f, "guaranteed-", "guaranteed", api.ResourceRequirements{
+				Limits: api.ResourceList{
```

Comment:
I think that in order to be classed as "guaranteed", you need to also specify `Requests` that are equal to the `Limits`.